fe/fe-core/src/main/java/org/apache/doris/analysis/GroupingInfo.java

```diff
@@ -89,7 +89,30 @@ public List<Expr> getPreRepeatExprs() {
     }
 
     public void substitutePreRepeatExprs(ExprSubstitutionMap smap, Analyzer analyzer) {
+        ArrayList<Expr> originalPreRepeatExprs = new ArrayList<>(preRepeatExprs);
         preRepeatExprs = Expr.substituteList(preRepeatExprs, smap, analyzer, true);
+
+        // remove unmaterialized slotRef from preRepeatExprs
+        ArrayList<Expr> materializedPreRepeatExprs = new ArrayList<>();
+        ArrayList<Expr> unMaterializedSlotRefs = new ArrayList<>();
+        for (int i = 0; i < preRepeatExprs.size(); ++i) {
+            Expr expr = preRepeatExprs.get(i);
+            if (expr instanceof SlotRef && !((SlotRef) expr).getDesc().isMaterialized()) {
+                unMaterializedSlotRefs.add(originalPreRepeatExprs.get(i));
+            } else {
+                materializedPreRepeatExprs.add(expr);
+            }
+        }
+        preRepeatExprs = materializedPreRepeatExprs;
+
+        // remove unmaterialized slotRef from outputTupleSmap and outputTupleDesc if there is any
+        for (Expr expr : unMaterializedSlotRefs) {
+            Expr rExpr = outputTupleSmap.get(expr);
+            outputTupleSmap.removeByRhsExpr(rExpr);
+            if (rExpr instanceof SlotRef) {
+                outputTupleDesc.getSlots().remove(((SlotRef) rExpr).getDesc());
```

Comment:
is it better to set it as not materialized instead of remove it?

Reply:
set unmaterialized is better, code is updated, thx